src/pcnt.rs

```diff
@@ -0,0 +1,543 @@
+use core::fmt::Debug;
+
+
+use esp_idf_sys::esp;
+use esp_idf_sys::pcnt_config_t;
+use esp_idf_sys::pcnt_set_mode;
+use esp_idf_sys::pcnt_unit_t;
+use esp_idf_sys::EspError;
+
+use bitflags::bitflags;
+
+use crate::gpio::AnyInputPin;
+use crate::gpio::Pin;
+
+type UnitHandle = pcnt_unit_t;
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum PcntChannel {
+    Channel0,
+    Channel1,
+}
+
+impl Into<esp_idf_sys::pcnt_channel_t> for PcntChannel {
```

Comment:
`pcnt_channel_t` smells to me as a peripheral that - if you are not careful - you can reuse at more than one place? If yes, then it needs to be modeled as a peripheral, not as a simple enum. Please consult the LEDC driver how this is done.

Reply:
@ivmarkov what if this was done in a similar manor to v5 and it was allocated from the unit, returning EspError(ESP_ERR_NO_MEM) if both have already been allocated?

Reply:
Let's not change the philosophy of the whole crate just yet. Let's try to model the V4 driver just the way everything else is modeled so far. We'll deal with V5 after that.

Reply:
So also implement the Unit as a peripheral?

Reply:
Right - now I understand your comment (a bit jet lagged) - channels are not peripherals.